benchmark/profile_restful_api.py

```diff
@@ -103,17 +105,28 @@ def _inference(self, req_queue: Queue, res_queue: Queue, session_id: int,
                     stream=stream_output,
                     session_id=session_id,
                     ignore_eos=True):
+                # Here we ignore the index of the multiple outputs and
+                # just put all of them together to compute tokens.
+                for choice in output.get('choices', []):
+                    if stream_output:
+                        full_output += choice['delta']['content']
+                    else:
+                        full_output += choice['message']['content']
                 timestamps.append(time.perf_counter())
 
             first_token_latency = np.round(timestamps[1] - timestamps[0], 3)
             token_latency = np.round(timestamps[-1] - timestamps[0], 3)
             # assert output.pop('finish_reason') == 'length', \
             #     f'Error. session_id({session_id}) request {output_seqlen} ' \
             #     f'tokens, but `finish_reason` is not `length`'
-            total_tokens = input_seqlen + output_seqlen
+            tokenlizer_start = time.perf_counter()
+            real_output_seqlen = len(self.tokenizer(full_output).input_ids)
```

Comment:
Encoding the text affects the inference performance.
I don't suggest doing that. 
If the output seqlen is needed, the server can return it.

Reply:
In stream mode, this statistic is not returned. 

In my tests(Qwen-72B-Chat), the tokenlizer takes only 0.027% of the whole benchmark elapsed time(tokenlizer speed: 77402.264 token/s for one concurrency), and the tokenlizer time has been removed in the final stats code.
```python
        stats = np.concatenate(stats).reshape(-1, 6)

        tokenlizer_time = np.sum(stats[:, 5], axis=0) / concurrency
        elapsed_time -= tokenlizer_time
```

Reply:
> Encoding the text affects the inference performance. I don't suggest doing that. If the output seqlen is needed, the server can return it.

lmdeploy doesn't support [`stream_options`](https://platform.openai.com/docs/api-reference/chat/create#chat-create-stream_options) to get stats from server in stream mode yet.